engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/FindTypeByFqnNode.java

```diff
@@ -0,0 +1,31 @@
+package org.enso.interpreter.node.expression.builtin.meta;
+
+import com.oracle.truffle.api.nodes.Node;
+import org.enso.interpreter.dsl.BuiltinMethod;
+import org.enso.interpreter.runtime.EnsoContext;
+import org.enso.interpreter.runtime.data.text.Text;
+import org.enso.pkg.QualifiedName;
+
+@BuiltinMethod(
+    type = "Meta",
+    name = "find_type_by_qualified_name",
+    description = "Finds a type by fully qualified name.",
+    autoRegister = false)
+public class FindTypeByFqnNode extends Node {
+  Object execute(Text fqn) {
+    var ctx = EnsoContext.get(this);
+    var fullName = QualifiedName.fromString(fqn.toString());
+    if (fullName.getParent().isDefined()) {
+      var moduleName = fullName.getParent().get();
+      for (var m : ctx.getTopScope().getModules()) {
+        if (m.getName().equals(moduleName)) {
```

Comment:
We are iterating all modules here. I imagine we can have a lot of them, given that every _file_ is a module (right?). Shouldn't we use a more efficient data structure?

The `getTopScope.getModules` delegates to `PackageRepository.getLoadedModules` which returns a sequence. But next to it, there is `PackageRepository.getModuleMap` which should allow us to get a mapping `modulename->module` that will resolve much more efficiently than this linear iteration.

Reply:
Done in 6be79f6be4